app/controllers/parent_interface/consent_forms/edit_controller.rb

```diff
@@ -39,14 +39,25 @@ def update
         )
       end
 
-      if current_step == :parent &&
-           @consent_form.parental_responsibility == "no"
-        return(
-          redirect_to session_parent_interface_consent_form_cannot_consent_responsibility_path(
-                        @session,
-                        @consent_form
-                      )
+      if current_step == :parent
+        if @consent_form.parental_responsibility == "no"
+          return(
+            redirect_to session_parent_interface_consent_form_cannot_consent_responsibility_path(
+                          @session,
+                          @consent_form
+                        )
+          )
+        end
+
+        # rename keys, taking parent_ out of the key
+        parent_params =
+          update_params
+            .except(:form_step)
+            .transform_keys { |key| key.to_s.gsub("parent_", "") }
+        (@consent_form.parent || @consent_form.build_parent).assign_attributes(
+          parent_params
         )
```

Comment:
Just for the sake of readability, could it be worth doing this:

```suggestion
        consent_object = @consent_form.parent || @consent_form.build_parent
        consent_object.assign_attributes(parent_params)
```

Reply:
Ya, this got refactored out as well in a later commit; now the parent is loaded or built in `EditController#set_parent` and just used where it's needed.